app/controllers/students_controller.rb

```diff
@@ -1,18 +1,24 @@
 require 'roo'
 class StudentsController < ApplicationController
+  before_action :set_student, only: [:show, :edit, :update]
+  before_action :authorize_user, only: [:index, :edit, :update, :destroy]
 
   def index
     @users = User.student
   end
 
   def show
-    @student = Student.find(params[:id])
     @user = @student.user
     authorize @student
   end
 
-  def new
-    @user = User.new
+  def edit
+  end
+
+  def update
+    params.permit!
```

Comment:
Please do not use `params.permit!` as it whitelist all the parameters and could lead to vulnerabilities. (https://stackoverflow.com/questions/52374655/ruby-on-rails-when-to-use-params-permit-and-how-to-replace-it)

Instead use strong parameters (https://edgeapi.rubyonrails.org/classes/ActionController/StrongParameters.html)

Reply:
Done